snuba/utils/streams/backends/kafka.py

```diff
@@ -617,24 +619,55 @@ def closed(self) -> bool:
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
 
 
+@dataclass
+class KafkaBrokerConfig:
+    config: Mapping[str, Any]
+
+
+def get_broker_config(bootstrap_servers: Sequence[str]) -> Mapping[str, Any]:
+    return {
+        "bootstrap.servers": ",".join(bootstrap_servers),
+    }
+
+
 def build_kafka_consumer_configuration(
-    bootstrap_servers: Sequence[str],
+    broker_config: Mapping[str, Any],
     group_id: str,
     auto_offset_reset: str = "error",
     queued_max_messages_kbytes: int = DEFAULT_QUEUED_MAX_MESSAGE_KBYTES,
     queued_min_messages: int = DEFAULT_QUEUED_MIN_MESSAGES,
-) -> Mapping[str, Any]:
-    return {
-        "enable.auto.commit": False,
-        "enable.auto.offset.store": False,
-        "bootstrap.servers": ",".join(bootstrap_servers),
-        "group.id": group_id,
-        "auto.offset.reset": auto_offset_reset,
-        # overridden to reduce memory usage when there's a large backlog
-        "queued.max.messages.kbytes": queued_max_messages_kbytes,
-        "queued.min.messages": queued_min_messages,
-        "enable.partition.eof": False,
-    }
+) -> KafkaBrokerConfig:
+    return KafkaBrokerConfig(
+        {
+            **broker_config,
+            "enable.auto.commit": False,
+            "enable.auto.offset.store": False,
+            "group.id": group_id,
+            "auto.offset.reset": auto_offset_reset,
+            # overridden to reduce memory usage when there's a large backlog
+            "queued.max.messages.kbytes": queued_max_messages_kbytes,
+            "queued.min.messages": queued_min_messages,
+            "enable.partition.eof": False,
+        }
+    )
+
+
+def build_kafka_producer_configuration(
+    broker_config: Mapping[str, Any],
+) -> KafkaBrokerConfig:
+    return KafkaBrokerConfig(
+        {
+            **broker_config,
+            "partitioner": "consistent",
+            "message.max.bytes": 50000000,  # 50MB, default is 1MB
```

Comment:
See my comment above, I would not hardcode parameters in this code and allow the methods to receive an override mapping.